test/e2e/containerruntimeconfig_test.go

```diff
@@ -0,0 +1,142 @@
+	"github.com/openshift/machine-config-operator/test/e2e/framework"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+)
+
+func TestContainerRuntimeConfigLogSizeMax(t *testing.T) {
+	runTestWithCtrcfg(t, "log-size-max", `log_size_max = (\S+)`, "30k", &mcfgv1.ContainerRuntimeConfiguration{
+		LogSizeMax: resource.MustParse("30k"),
+	})
+}
+
+func TestContainerRuntimeConfigPidsLimit(t *testing.T) {
+	runTestWithCtrcfg(t, "pids-limit", `pids_limit = (\S+)`, "12345", &mcfgv1.ContainerRuntimeConfiguration{
+		PidsLimit: 12345,
+	})
+}
+
+func runTestWithCtrcfg(t *testing.T, testName, regexKey, expectedConfValue string, cfg *mcfgv1.ContainerRuntimeConfiguration) {
+	cs := framework.NewClientSet("")
+	testUUID := uuid.NewUUID()
+	matchValue := fmt.Sprintf("%s-%s", testName, testUUID)
+	ctrcfgName := fmt.Sprintf("ctrcfg-%s", testUUID)
+
+	// label one node from the pool to specify which worker to update
+	unlabelFunc := labelRandomNodeFromPool(t, cs, "worker", mcpNameToRole(matchValue))
+	defer unlabelFunc()
+
+
+	// check that the node properly updated
+	nodes, err := getNodesByRole(cs, matchValue)
+	require.Nil(t, err)
+	require.Len(t, nodes, 1)
+	infraNode := nodes[0]
+
+	oldConfValue := getValueFromCrioConfig(t, infraNode.Name, regexKey)
+	if oldConfValue == expectedConfValue {
+		t.Logf("default configuration value %s same as value being tested against. Consider updating the test", oldConfValue)
+		return
+	}
+
+	// create an MCP to match the node we tagged
+	mcpCleanupFunc := createMCP(t, cs, matchValue)
+	defer mcpCleanupFunc()
+
+	// create old mc to have something to check we roll back to
+	oldMCConfig := createMC(matchValue, matchValue)
+	_, err = cs.MachineConfigs().Create(oldMCConfig)
+	require.Nil(t, err)
+
+	defer func() {
+		err := cs.MachineConfigs().Delete(matchValue, &metav1.DeleteOptions{})
+		require.Nil(t, err)
+	}()
+
+	oldRenderedMC, err := waitForRenderedConfig(t, cs, matchValue, matchValue)
+	require.Nil(t, err, "failed to render MC")
+
+	// wait for the ctrcfg to complete its update
+	err = waitForPoolComplete(t, cs, matchValue, oldRenderedMC)
+	require.Nil(t, err)
+
+	// create our ctrcfg and attach it to our created node pool
+	cleanupCtrcfgFunc := createCtrcfgWithConfig(t, cs, ctrcfgName, matchValue, cfg)
+
+	// wait for the ctrcfg to show up
+	mcName, err := waitForRenderedConfigFromCtrcfg(t, cs, matchValue, ctrcfgName)
+	require.Nil(t, err, "failed to render machine config from container runtime config")
+
+	// wait for the ctrcfg to complete its update
+	err = waitForPoolComplete(t, cs, matchValue, mcName)
+	require.Nil(t, err)
+
+	// verify value was changed
+	newConfValue := getValueFromCrioConfig(t, infraNode.Name, regexKey)
+	assert.Equal(t, newConfValue, expectedConfValue)
```

Comment:
maybe a message as an arg here?

Reply:
good idea, I should do that on all of them probably